### PR TITLE
support-android-test component + "since" parameter for feature-session-bundling

### PR DESCRIPTION
**Pull request by pocmo**
### Pull Request checklist

- [x] **Quality**: This PR builds and passes detekt/ktlint checks (A pre-push hook is recommended)
- [x] **Tests**: This PR includes thorough tests or an explanation of why it does not
- [x] **Changelog**: This PR includes [a changelog entry](https://github.com/mozilla-mobile/android-components/blob/master/docs/changelog.md) or does not need one
- [x] **Accessibility**: The code in this PR follows [accessibility best practices](https://github.com/mozilla-mobile/shared-docs/blob/master/android/accessibility_guide.md) or does not include any user facing features

### CODE REVIEW

File: components/feature/session-bundling/src/androidTest/java/mozilla/components/feature/session/bundling/SessionBundleStorageTest.kt [hunk 1]
```diff
@@ -0,0 +1,103 @@
...
+            assertTrue(urls.contains("https://www.firefox.com"))
+        }
+    }
+}
+
+private fun List<BundleEntity>.toUrlList(): List<String> =
+    this
+        .map { bundle -> bundle.urls }
+        .map { urls ->
+            assertEquals(1, urls.entries.size)
```
Comment (Amejia481) [inline components/feature/session-bundling/src/androidTest/java/mozilla/components/feature/session/bundling/SessionBundleStorageTest.kt]: nit: You're doing an assertion in this util function ,intentionally?

File: components/support/android-test/build.gradle [hunk 2]
```diff
@@ -0,0 +1,40 @@
...
+            minifyEnabled false
+            proguardFiles getDefaultProguardFile('proguard-android.txt')
+        }
+    }
+
+    lintOptions {
+        // Disabled since this caused issues with Android Gradle Plugin 3.2.1+ (NullPointerException:InvalidPackageDetector)
+        tasks.lint.enabled = false
+
+        lintConfig file("lint.xml")
```
Comment (Amejia481) [inline components/support/android-test/build.gradle]: 🤔  If we are disabling lint why do we have to pass a `lintConfig` file?

File: components/support/android-test/src/main/java/mozilla/components/support/android/test/LiveData.kt [hunk 3]
```diff
@@ -0,0 +1,40 @@
...
+
+/**
+ * Subscribes to the [LiveData] object and blocks until a value was observed. Returns the value or throws
+ * [InterruptedException] if no value was observed in the given [timeout] (using [unit]).
+ *
+ * This is helpful for tests using [LiveData] objects that won't contain any data unless observed (e.g. LiveData objects
+ * returned by Room).
+ */
+@Throws(InterruptedException::class)
+fun <T> LiveData<T>.awaitValue(
```
Comment (Amejia481) [inline components/support/android-test/src/main/java/mozilla/components/support/android/test/LiveData.kt]: Nice stuff!

File: components/feature/session-bundling/src/main/java/mozilla/components/feature/session/bundling/SessionBundleStorage.kt [hunk 4]
```diff
@@ -130,12 +130,16 @@ class SessionBundleStorage(
 
     /**
      * Returns the last saved [SessionBundle] instances (up to [limit]) as a [LiveData] list.
+     *
+     * @param since (Optional) Unix timestamp (milliseconds). If set this method will only return [SessionBundle]
+     * instances that have been saved since the given timestamp.
+     * @param limit (Optional) Maximum number of [SessionBundle] instances that should be returned.
      */
-    fun bundles(limit: Int = 20): LiveData<List<SessionBundle>> {
+    fun bundles(since: Long = 0, limit: Int = 20): LiveData<List<SessionBundle>> {
```
Comment (Amejia481) [inline components/feature/session-bundling/src/main/java/mozilla/components/feature/session/bundling/SessionBundleStorage.kt]: fun fact:  If [I convert 0 to a date](http://timestamp.online/) is `12/31/1969, 7:00:00 PM` luckily all our bundles are going to be newer than that 😆

File: components/feature/session-bundling/src/androidTest/java/mozilla/components/feature/session/bundling/SessionBundleStorageTest.kt [hunk 5]
```diff
@@ -0,0 +1,103 @@
...
+        bundleDao = database.bundleDao()
+    }
+
+    @After
+    fun tearDown() {
+        database.close()
+    }
+
+    @Test
+    fun testInsertingAndReadingBundles() {
```
[the remark this answers is not in the document]
Comment (pocmo) [reply components/feature/session-bundling/src/androidTest/java/mozilla/components/feature/session/bundling/SessionBundleStorageTest.kt]: Yeah, unfortunately Android doesn't support blanks in the methods names. That only works locally for tests that run on the JVM :(

[the remark this answers is not in the document]
Comment (Amejia481) [reply components/feature/session-bundling/src/androidTest/java/mozilla/components/feature/session/bundling/SessionBundleStorageTest.kt]: 👍

File: components/feature/session-bundling/src/main/java/mozilla/components/feature/session/bundling/SessionBundleStorage.kt (hunk 4 above)
Comment (pocmo) [reply components/feature/session-bundling/src/main/java/mozilla/components/feature/session/bundling/SessionBundleStorage.kt]: Yeah, timezones are hard :D

**Comment by csadilek** [comment]
Landing this to get the release-prep rebased. We can address nits as a follow-up if needed.

**Pull request merged**